### PR TITLE
[RQLY-190] Added metadata in Requestly UI

### PR DESCRIPTION
**Pull request by Samanvay-Arya**
[RQLY-190] Added Device and SDK Metadata in Requestly UI

[RQLY-190]: https://requestlyio.atlassian.net/browse/RQLY-190?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: requestly-android-okhttp/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -67,4 +67,9 @@
     <string name="rq_interceptor_unique_device_id_key">UNIQUE_DEVICE_ID</string>
     <string name="rq_interceptor_capture_enabled">CAPTURE_ENABLED</string>
     <string name="rq_interceptor_settings">Setting</string>
+    <string name="rq_interceptor_more_details">More Details</string>
+    <string name="rq_interceptor_device_id">Device Id</string>
```
Comment (wrongsahil) [inline requestly-android-okhttp/src/main/res/values/strings.xml:71]:
```
<string name="rq_interceptor_device_id_label">Device Id</string>
<string name="rq_interceptor_copy_device_id_desc">Copy Device Id</string>
<string name="rq_interceptor_app_id_label">App Id</string>
```

File: requestly-android-okhttp/src/main/res/menu/rq_interceptor_transaction.xml [hunk 2]
```diff
@@ -8,6 +8,13 @@
         android:visible="false"
         app:actionViewClass="androidx.appcompat.widget.SearchView"
         app:showAsAction="collapseActionView|ifRoom" />
+    <item
+        android:title="@string/rq_interceptor_more_details"
+        android:id="@+id/more_Details"
+        android:icon="@drawable/rq_interceptor_ic_more_info"
+        android:visible="false"
+        app:showAsAction="ifRoom"
+        />
```
Comment (wrongsahil) [inline requestly-android-okhttp/src/main/res/menu/rq_interceptor_transaction.xml:17]: No need to add this here.

File: requestly-android-okhttp/src/main/res/layout/rq_interceptor_more_info_dialog_layout.xml [hunk 3]
```diff
@@ -0,0 +1,106 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:paddingVertical="20dp"
+    android:layout_height="wrap_content">
+
+    <TextView
+        android:id="@+id/rq_interceptor_textview"
```
Comment (wrongsahil) [inline requestly-android-okhttp/src/main/res/layout/rq_interceptor_more_info_dialog_layout.xml:9]: Name these views properly
Eg: `@+id/rq_interceptor_device_id_label`, `@+id/rq_interceptor_device_id` ...

File: requestly-android-okhttp/src/main/res/layout/rq_interceptor_more_info_dialog_layout.xml [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        app:layout_constraintEnd_toStartOf="@+id/rq_interceptor_DeviceID_Copy_ImageButton"
+        app:layout_constraintHorizontal_bias="0.5"
+        app:layout_constraintStart_toEndOf="@+id/rq_interceptor_textview"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintVertical_bias="0.499"
+        app:layout_constraintWidth_default="percent"
+        app:layout_constraintWidth_percent="0.4" />
+
+    <ImageButton
+        android:id="@+id/rq_interceptor_DeviceID_Copy_ImageButton"
```
Comment (wrongsahil) [inline requestly-android-okhttp/src/main/res/layout/rq_interceptor_more_info_dialog_layout.xml:39]: `@+id/rq_interceptor_device_id_copy_button`

File: requestly-android-okhttp/src/main/res/layout/rq_interceptor_more_info_dialog_layout.xml [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintWidth_default="percent"
+        app:layout_constraintWidth_percent="0.3" />
+
+    <TextView
+        android:id="@+id/rq_interceptor_DeviceId_TextView"
+        style="@style/RqInterceptor.TextAppearance.Title"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:text="@string/rq_interceptor_excess_lime"
```
Comment (wrongsahil) [inline requestly-android-okhttp/src/main/res/layout/rq_interceptor_more_info_dialog_layout.xml:28]: Why are we using `excess_lime` as default? We should show `Not Initialized` or `null` instead

File: requestly-android-okhttp/src/main/res/layout/rq_interceptor_more_info_dialog_layout.xml [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+        app:layout_constraintWidth_default="percent"
+        app:layout_constraintWidth_percent="0.3" />
+
+    <TextView
+        android:layout_marginTop="15dp"
+        android:id="@+id/rq_interceptor_AppId_TextView"
+        style="@style/RqInterceptor.TextAppearance.Title"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:text="@string/rq_interceptor_excess_lime"
```
Comment (wrongsahil) [inline requestly-android-okhttp/src/main/res/layout/rq_interceptor_more_info_dialog_layout.xml:78]: Same here. We should show `Not Initialized` or `null` as default value here

File: requestly-android-okhttp/src/main/kotlin/io/requestly/android/okhttp/internal/ui/MainActivity.kt [hunk 7]
```diff
@@ -172,6 +186,36 @@ internal class MainActivity :
...
+
+        deviceIdTV.text=RQClientProvider.client().deviceId
+        appIdTV.text=SettingsManager.getInstance().getAppToken()
+
+//                appIdTV.setText(rqClientPro.sdkId.toString())
+        deviceIdCopyIB.setOnClickListener {
+            Toast.makeText(this@MainActivity,"Device Id Copied!",Toast.LENGTH_SHORT).show()
+            val clipboardManager = getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
+            val clipData = ClipData.newPlainText("Device Id", android.os.Build.DEVICE)
+            clipboardManager.setPrimaryClip(clipData)
```
Comment (wrongsahil) [inline requestly-android-okhttp/src/main/kotlin/io/requestly/android/okhttp/internal/ui/MainActivity.kt:210]: Abstract this in a separate method

File: requestly-android-okhttp/src/main/kotlin/io/requestly/android/okhttp/internal/ui/MainActivity.kt [hunk 8]
```diff
@@ -172,6 +186,36 @@ internal class MainActivity :
         positiveButtonText = getString(R.string.rq_interceptor_export),
         negativeButtonText = getString(R.string.rq_interceptor_cancel)
     )
+    private fun getInfoDialog(){
+        val view= View.inflate(this@MainActivity,    R.layout.rq_interceptor_more_info_dialog_layout,null)
+        val builder= AlertDialog.Builder(this@MainActivity)
+        builder.setView(view)
+        val dialog=builder.create()
+        dialog.show()
+        dialog.setCancelable(true)
```
Comment (wrongsahil) [inline requestly-android-okhttp/src/main/kotlin/io/requestly/android/okhttp/internal/ui/MainActivity.kt:195]: Abstract this in `createInfoDialog` Method

File: requestly-android-okhttp/src/main/kotlin/io/requestly/android/okhttp/internal/ui/MainActivity.kt [hunk 9]
```diff
@@ -172,6 +186,36 @@ internal class MainActivity :
...
+
+//                appIdTV.setText(rqClientPro.sdkId.toString())
+        deviceIdCopyIB.setOnClickListener {
+            Toast.makeText(this@MainActivity,"Device Id Copied!",Toast.LENGTH_SHORT).show()
+            val clipboardManager = getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
+            val clipData = ClipData.newPlainText("Device Id", android.os.Build.DEVICE)
+            clipboardManager.setPrimaryClip(clipData)
+        }
+        appIdCopyIB.setOnClickListener {
+            Toast.makeText(this@MainActivity,"App Id is copied!",Toast.LENGTH_SHORT).show()
```
Comment (wrongsahil) [inline requestly-android-okhttp/src/main/kotlin/io/requestly/android/okhttp/internal/ui/MainActivity.kt:213]: Abstract this in `addInfoDialogListeners`

**Review by wrongsahil — CHANGES_REQUESTED** [review]

File: gradle.properties [hunk 10]
```diff
@@ -9,7 +9,7 @@
 
 # Specifies the JVM arguments used for the daemon process.
 # The setting is particularly useful for tweaking memory settings.
-org.gradle.jvmargs=-Xmx1536m --illegal-access=permit
```
[the remark this answers is not in the document]
Comment (Samanvay-Arya) [reply gradle.properties:12]: gradle was throwing some error

**Pull request closed without merging**